src/config/language.c

```diff
@@ -37,10 +41,27 @@ void CONFIG_ReadLang(u8 idx) {(void)idx;}
 void CONFIG_EnableLanguage(int state) {(void)state;}
 #else
 static u16 fnv_16_str(const char *str);
-static char strings[8192];
 static u16 table_size;
-#define MAX_STRINGS 475
+static unsigned fix_crlf(char *str);
+
+#ifndef MAX_LINE
 #define MAX_LINE 300
+#endif
+
+#ifndef MAX_STRINGS
+#define MAX_STRINGS 475
+#endif
+
+#if !SUPPORT_DYNAMIC_LOCSTR
+    static char strings[8192];
+#else
+    #define MAX_STRING_BUFFER 128
+
```

Comment:
We need to update the ../utils/check_string_size.pl to understand this new logic somehow.
It needs to get the proper value for MAX_LINE and to know whether the we're in cache mode or not.
Also, I'm not sure how to properly calculate the MAX_STRING_BUFFER value since it may consist of multiple strings.

Reply:
Seems no good automatical way to calculate it properly. 128 just a guess as all cases so far which are using multi _trs with a relative short string.

git grep sprintf | grep _tr
git grep snprintf | grep _tr

Reply:
For now, MAX_STRING_BUFFER is just a best guess. we can enlarge it if we have more memory.
this is current state:
ROM: 0x08003000 - 0x0801f754 = 113.83kB
RAM: 0x20000000 - 0x20002f9c =  11.90kB
STK: 0x20002f9c - 0x20004000 =   4.10kB